server/src/main/java/io/crate/expression/scalar/ExtractFunctions.java

```diff
@@ -72,6 +77,16 @@ public class ExtractFunctions {
         entry(SECOND, ISOChronology.getInstanceUTC().secondOfMinute())
     );
 
+    private static final Map<Extract.Field, Function<Period, Integer>> INTERVAL_FIELDS_MAP_WITH_INT_RETURN = Map.ofEntries(
+        entry(YEAR, p -> p.get(DurationFieldType.years())),
+        entry(QUARTER, p -> p.get(DurationFieldType.months()) / 4),
+        entry(MONTH, p -> p.get(DurationFieldType.months())),
+        entry(DAY, p -> p.get(DurationFieldType.days())),
+        entry(HOUR, p -> p.get(DurationFieldType.hours())),
+        entry(MINUTE, p -> p.get(DurationFieldType.minutes())),
+        entry(SECOND, p -> p.get(DurationFieldType.seconds()))
+    );
+
```

Comment:
Could define this inline in `register` so that it can be garbage collected. Otherwise this unnecessarily occupies memory during the lifetime of a node.